src/test/java/com/bobocode/svydovets/ioc/core/beanDefinitionFactory/BeanDefinitionFactoryTest.java

```diff
@@ -1,40 +1,83 @@
+
     @BeforeEach
     void setUp() {
+        beanDefinitionFactory = new BeanDefinitionFactory();
     }
 
     @AfterEach
     void tearDown() {
     }
 
     @Test
-    void registerBeanDefinitions() {
+    void shouldRegisterComponentAnnotationBeanDefinitions() {
+        beanDefinitionFactory.registerBeanDefinitions(Set.of(CommonService.class, MessageService.class));
+        BeanDefinition commonServiceBeandefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("commonService");
+        BeanDefinition messageServiceBeanDefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("messageService");
+        assertThat(commonServiceBeandefinition.getClass()).isEqualTo(ComponentAnnotationBeanDefinition.class);
+        assertThat(messageServiceBeanDefinition.getClass()).isEqualTo(ComponentAnnotationBeanDefinition.class);
+        assertThat(commonServiceBeandefinition).isNotNull();
+        assertThat(messageServiceBeanDefinition).isNotNull();
     }
 
     @Test
-    void registerBeanDefinition() {
+    void shouldRegisterComponentAnnotationBeanDefinition() {
+        beanDefinitionFactory.registerBeanDefinition(CommonService.class);
+        BeanDefinition commonServiceBeanDefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("commonService");
+        assertThat(commonServiceBeanDefinition.getClass()).isEqualTo(ComponentAnnotationBeanDefinition.class);
+        assertThat(commonServiceBeanDefinition).isNotNull();
     }
 
     @Test
-    void getBeanDefinitionByBeanName() {
+    void shouldReturnBeanFromConfigClassWith() {
+        beanDefinitionFactory.registerBeanDefinitions(Set.of(CommonService.class, BasePackageWithAdditionalBeansConfig.class));
+        BeanDefinition trimServiceBeanDefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("megaTrimService");
+        String scope = trimServiceBeanDefinition.getScope();
+        assertThat(trimServiceBeanDefinition.getClass()).isEqualTo(BeanAnnotationBeanDefinition.class);
+        assertThat(trimServiceBeanDefinition).isNotNull();
+        assertThat(scope).isEqualTo("singleton");
     }
 
     @Test
-    void createBeanDefinitionMapByConfigClass() {
+    void shouldReturnCollectDefinitionFromServiceWithConstructorInjection() {
+        beanDefinitionFactory.registerBeanDefinition(ConstructorInjectionService.class);
+        ComponentAnnotationBeanDefinition serviceDefinition = (ComponentAnnotationBeanDefinition) beanDefinitionFactory.getBeanDefinitionByBeanName("constructorInjectionService");
+        assertThat(serviceDefinition.getClass()).isEqualTo(ComponentAnnotationBeanDefinition.class);
+        assertThat(serviceDefinition).isNotNull();
+        assertThat(serviceDefinition.getScope()).isEqualTo("singleton");
+        assertThat(serviceDefinition.getInitializationConstructor()).isNotNull();
+        assertThat(serviceDefinition.getInitializationConstructor().getParameterTypes()[0]).isEqualTo(InjectionCandidate.class);
```

Comment:
We can split this test into 2. The first one will check that bean definition is created (line 70) and has the correct class type (line 69). The second one will check that all fields is filled correctly. 
Pay attention that you perform cast (line 68) before checking the class type (line 69)